cupy/core/include/cupy/math_constants.h

```diff
@@ -0,0 +1,110 @@
+#ifndef CUPY_MATH_CONSTANTS_H
```

Comment:
The backport should probably use the more minimal version of this file that was introduced for #4565. 

#4565 also helped avoid a patch during the conda package build

Reply:
Yeah, we need to backport that one first :)

Reply:
Good catch, @grlee77!

Reply:
Perhaps this PR can be backported first, and then followed by backporting #4565?